src/parquet/schema.cc

```diff
@@ -620,6 +620,15 @@ const ColumnDescriptor* SchemaDescriptor::Column(int i) const {
   return &leaves_[i];
 }
 
+int SchemaDescriptor::ColumnIndex(const NodePtr& node) const {
```

Comment:
@itaiin It occurred to me this morning, we shouldn't use shared_ptr as keys as they are mutable. And we don't need to: we can use ColumnPath of NodePtr as keys. And it's fair easy to reconstruct ColumnPath from NodePtr